src/composables/usePanels.ts

```diff
@@ -0,0 +1,42 @@
+import { computed, ref, Ref, watch } from '@vue/composition-api';
+
+export interface Panel {
+  key: string;
+  isOpen: boolean;
+}
+
+export function usePanels(keys: Ref<Set<string>>) {
+  const panels = ref<Panel[]>([]);
+
+  watch(
+    keys,
+    (newKeys, oldKeys) => {
+      // remove deleted
+      panels.value = panels.value.filter(({ key }) => newKeys.has(key));
+
+      const addedKeys = new Set(
+        [...newKeys].filter((key) => !oldKeys?.has(key))
+      );
+      // order in  temporal created order, not template order.  See https://github.com/vuetifyjs/vuetify/issues/11225
+      panels.value = [
+        ...panels.value,
+        ...[...addedKeys].map((key) => ({ key, isOpen: true })),
+      ];
+    },
+    { flush: 'post', immediate: true }
```

Comment:
What's the purpose of `flush: 'post'` here? Is it so that we can apply the correct panel value on the expansion panel after it has updated its DOM?

Reply:
Added a comment there:
```
// flush: 'post' keeps panels open when deleting patients
// because panels.value has the same number panels in v-expansion-panels after flush
```